syft/serde/msgpack/torch_serde.py

```diff
@@ -148,6 +148,9 @@ def _simplify_torch_tensor(worker: AbstractWorker, tensor: torch.Tensor) -> bin:
     # and here... leaving a reerence here so i can find it later
     # TODO fix pointer bug
 
+    sender = tensor.sender if hasattr(tensor, "sender") else None
```

Comment:
To be sure I understand, you added a sender and an origin_id attribute on an object when it's sent?
I didn't think as much as you about it but maybe it would be cleaner to add this information in the message and not add it on the object?

Reply:
Haha it was in the message but based on Karl suggestions I moved it to the object, where it seems somewhat cleaner :)

Reply:
Hum, I still don't think dynamically adding attributes to object is clean but if you agreed on it, it's ok for now :)

Reply:
I'm not a huge fan of dynamically adding attributes either. My thinking was that we're essentially trying to build up a cross-worker tensor chain, and tensors already contain the information required to handle normal tensor chains, so that might be a reasonable place to keep this information too. It does seem like there's some overlap between pointers, messages/actions, and tensor chaining that's pretty fuzzy right now.

I'm not sure `sender` is actually the right name for this attribute if it's on a tensor, because that sort of implies a messaging context. Maybe `origin`? 🤔

Reply:
I'll go for `origin` and  set them in the "constructor" (or kindof) :)